### PR TITLE
Add more verbose output from pytest suite

### PR DESCRIPTION
**Pull request by phracek**
This pull request only adds more verbose output from pytest module.

We would like to see what's happened during the tests.

### CODE REVIEW

**Comment by phracek** [comment]
[test-openshift-pytest]

**Comment by phracek** [comment]
The output looks great. See https://artifacts.osci.redhat.com/testing-farm/8a939e48-ed7d-4491-83c2-ae79c8620f39/. We can see progress of the tests and 'short test summary info' as well.

```
=========================== short test summary info ============================
PASSED test_deploy_templates.py::TestDeployTemplate::()::test_python_template_inside_cluster[django-postgresql.json]
PASSED test_deploy_templates.py::TestDeployTemplate::()::test_python_template_inside_cluster[django-postgresql-persistent.json]
PASSED test_imagestreams_quickstart.py::TestImagestreamsQuickstart::()::test_python_template_inside_cluster[django-postgresql.json]
PASSED test_imagestreams_quickstart.py::TestImagestreamsQuickstart::()::test_python_template_inside_cluster[django-postgresql-persistent.json]
PASSED test_python_ex_standalone.py::TestPythonExTemplate::()::test_python_ex_template_inside_cluster
PASSED test_python_ex_template.py::TestPythonExTemplate::()::test_python_ex_template_inside_cluster
```

**Comment by phracek** [comment]
In case there is something, that I have missing from PyTest arguments, improvements are welcome, as commented in this pull request.

File: test/run-openshift-pytest [hunk 1]
```diff
@@ -8,4 +8,4 @@
 
 THISDIR=$(dirname ${BASH_SOURCE[0]})
 
-cd "${THISDIR}" && python3 -m pytest --showlocals -vv test_*.py
+cd "${THISDIR}" && python3 -m pytest -s -rxfapP --showlocals -vv test_*.py
```
Comment (frenzymadness) [inline test/run-openshift-pytest:11]: Could you please explain this part: `-rxfapP`

I see this in the help text of pytest:

```
  -r chars              Show extra test summary info as specified by chars: (f)ailed,
                        (E)rror, (s)kipped, (x)failed, (X)passed, (p)assed, (P)assed with
                        output, (a)ll except passed (p/P), or (A)ll. (w)arnings are enabled
                        by default (see --disable-warnings), 'N' can be used to reset the
                        list. (default: 'fE').
```

If I understand it correctly, the `a` means `all except passed` and then you have `pP` to include passed which basically means everything. Can this be replaced by `-rA`?.

Comment (phracek) [reply test/run-openshift-pytest]: For some kind of reason, it does not work. But now, it is working. https://github.com/sclorg/s2i-python-container/pull/694/commits/5d908f097e966eac3f8053e137b2fd5a67547976

**Comment by phracek** [comment]
Le't run all tests. For sure.

[test-all]

File: test/run-openshift-pytest (hunk 1 above)
Comment (phracek) [reply test/run-openshift-pytest]: @frenzymadness Well, It looks like extra summary info is not shown. See test results here: https://artifacts.osci.redhat.com/testing-farm/b786a35c-ed0a-457e-a35a-6e622cf2a4da/

Comment (phracek) [reply test/run-openshift-pytest]: Locally I can see this:
```
=============================================================================================== short test summary info ===============================================================================================
PASSED test_deploy_templates.py::TestDeployTemplate::test_python_template_inside_cluster[django-postgresql.json]
PASSED test_deploy_templates.py::TestDeployTemplate::test_python_template_inside_cluster[django-postgresql-persistent.json]
PASSED test_imagestreams_quickstart.py::TestImagestreamsQuickstart::test_python_template_inside_cluster[django-postgresql.json]
PASSED test_imagestreams_quickstart.py::TestImagestreamsQuickstart::test_python_template_inside_cluster[django-postgresql-persistent.json]
PASSED test_python_ex_standalone.py::TestPythonExTemplate::test_python_ex_template_inside_cluster
PASSED test_python_ex_template.py::TestPythonExTemplate::test_python_ex_template_inside_cluster
```

**Comment by phracek** [comment]
@frenzymadness It is interesting issue: `Testing Farm - RHEL8 - PyTest - OpenShift 4 - 3.11` and in logs https://artifacts.osci.redhat.com/testing-farm/b786a35c-ed0a-457e-a35a-6e622cf2a4da/ at the end is no 'summary info'
But in `Testing Farm - RHEL9 - PyTest - OpenShift 4 - 3.11` and in logs https://artifacts.osci.redhat.com/testing-farm/866f4180-ce8f-47b2-a65d-b6fc245897dc/ at the end is summary:

```
=========================== short test summary info ============================
PASSED test_deploy_templates.py::TestDeployTemplate::test_python_template_inside_cluster[django-postgresql.json]
PASSED test_deploy_templates.py::TestDeployTemplate::test_python_template_inside_cluster[django-postgresql-persistent.json]
PASSED test_imagestreams_quickstart.py::TestImagestreamsQuickstart::test_python_template_inside_cluster[django-postgresql.json]
PASSED test_imagestreams_quickstart.py::TestImagestreamsQuickstart::test_python_template_inside_cluster[django-postgresql-persistent.json]
PASSED test_python_ex_standalone.py::TestPythonExTemplate::test_python_ex_template_inside_cluster
PASSED test_python_ex_template.py::TestPythonExTemplate::test_python_ex_template_inside_cluster
================== 6 passed, 4 warnings in 651.28s (0:10:51) ===================
```

**Comment by frenzymadness** [comment]
If you use main Python (3.6 in RHEL 8, 3.9 in RHEL 9) it might also lead to different pytest versions being installed due to newest pytest not supporting Python 3.6.

**Comment by pkubatrh** [comment]
Let's move to latest Python and use rpm-based pytest if possible on all RHEL versions.

**Comment by phracek** [comment]
Run the tests as soon as this MR is merged https://gitlab.cee.redhat.com/platform-eng-core-services/sclorg-tmt-plans/-/merge_requests/205

**Comment by phracek** [comment]
Python-3.12 tests are here: https://github.com/sclorg/s2i-ruby-container/pull/551 and especially here: http://artifacts.osci.redhat.com/testing-farm/eb7859e2-aadc-4982-875c-a7de06d22597

**Comment by phracek** [comment]
[test-openshift-pytest]

**Review by frenzymadness — APPROVED** [review]

**Pull request merged**